sycl/include/sycl/ext/intel/experimental/esimd/memory.hpp

```diff
@@ -3034,7 +3049,10 @@ template <__ESIMD_NS::atomic_op Op, typename T, int N,
           cache_hint L1H = cache_hint::none, cache_hint L3H = cache_hint::none,
           typename AccessorTy>
 __ESIMD_API std::enable_if_t<
-    sycl::detail::acc_properties::is_local_accessor_v<AccessorTy>,
+    __ESIMD_DNS::is_local_accessor_with_v<
+        AccessorTy, __ESIMD_DNS::accessor_mode_cap::can_write> &&
+        __ESIMD_DNS::is_local_accessor_with_v<
+            AccessorTy, __ESIMD_DNS::accessor_mode_cap::can_read>,
```

Comment:
nit: If create is_rw_load_accessor_v, then this code could be much shorter:
```suggestion
    __ESIMD_DNS::is_rw_local_accessor_v<AccessorTy>,
```